### PR TITLE
Improve loader

### PR DESCRIPTION
**Pull request by privat**
This PR improves the loader that can now accepts

* directories of projects: to load all modules of all projects (eg `contrib`)
* root directories of groups: to load all modules of the groups (eg `contrib/benitlux`).
  Previously, this worked if there was no `src` subdirectory)

This works for tools that rely on full_parse (basically all tools except the engines).

~~~sh
$ ./test_test_phase ../lib -q
It works
I have 89 projects
I have 285 modules
I have 1038 classes
For 1366 definitions of classes
I have 7101 methods
For 9098 definitions of methods
~~~

Note that if any files has an error most tools will just stop.

### CODE REVIEW

File: src/loader.nit [hunk 1]
```diff
@@ -108,7 +108,7 @@ redef class ModelBuilder
 	end
 
 	# Load a bunch of modules and groups.
-	# Each name can be a module or a group.
+	# Each name can be a module or a group or a directory of projects.
```
Comment (Morriar) [inline src/loader.nit]: Is this a name or a path?

**Comment by Morriar** [comment]
+1

**Comment by privat** [comment]
improved the documentation of `parse_full`. It this better?

**Comment by xymus** [comment]
:shipit:

**Pull request merged**